src/slots/named-manual.js

```diff
@@ -5,104 +5,53 @@
  * - React to dynamic slot changes (added slots, removed slots, renamed slots)
  *
  */
-import SlotObserver from "./slot-observer.js";
 
-let mutationObserver;
+import base, { slots } from "./base.js";
 
-export function getSlot (host, child, slots) {
-	let slotName = child.slot;
+export const dependencies = [base];
 
-	if (slotName) {
-		slots[slotName] ??= host.shadowRoot.querySelector(`slot[name="${slotName}"]`);
-		return slots[slotName];
+export const slottedObserver = new MutationObserver(records => {
+	let host = records[0].target;
+	if (host.nodeType !== Node.ELEMENT_NODE) {
+		host = host.parentNode;
 	}
 
-	// Default slot
-	return slots[""];
-}
-
-export function assign (child, slots) {
-	let slot = getSlot(this, child, slots);
-	if (slot) {
-		let isAssigned = slot.assignedNodes().includes(child);
-
-		if (!isAssigned) {
-			slot.assign(child);
-			slot.dispatchEvent(new Event("slotchange"), { bubbles: true });
+	for (let r of records) {
+		if (r.type === "attributes") {
+			host[slots].assign(r.target);
 		}
-	}
-}
-
-export function unassign (child, slots) {
-	let slot = getSlot(this, child, slots);
-	if (slot) {
-		const assignedNodes = slot.assignedNodes();
-		let isAssigned = assignedNodes.includes(child);
-
-		if (isAssigned) {
-			slot.assign(...assignedNodes.filter(node => node !== child));
-			slot.dispatchEvent(new Event("slotchange"), { bubbles: true });
+		else {
+			for (let node of r.addedNodes) {
+				host[slots].assign(node);
+			}
 		}
 	}
-}
-
-export function slotsChanged (records) {
-	for (let r of records) {
-		// TODO
-	}
-}
-
-export default function  (Class, options = {}) {
-	return {
-		init () {
-			if (this.shadowRoot?.slotAssignment !== "manual") {
-				// Nothing to do here
-				return;
-			}
-
-			// slotchange won’t fire in this case, so we need to do this the old-fashioned way
-			mutationObserver ??= new MutationObserver(mutations => {
-				let slots = {};
-
-				let nodesToAssign = records.flatMap(r =>
-					r.type === "attributes" ? [r.target] : r.addedNodes);
-				let nodesToUnassign = records.flatMap(r =>
-					r.type === "attributes" ? [] : r.removedNodes);
+});
 
-				for (let node of nodesToAssign) {
-					assign(node, slots);
-				}
-
-				for (let node of nodesToUnassign) {
-					unassign(node, slots);
-				}
-			});
+export const hooks = {
+	connected () {
+		if (this[slots].shadowRoot?.slotAssignment !== "manual") {
```

Comment:
```suggestion
		if (this[slots]?.shadowRoot?.slotAssignment !== "manual") {
```

In my experiments, for elements without slots, this line might throw an error, so I'd make it a bit more error-proof.